share/picongpu/examples/LaserWakefield/etc/picongpu/0004gpus_gui.cfg

```diff
@@ -0,0 +1,86 @@
+# Copyright 2013-2017 Axel Huebl, Rene Widera, Felix Schmitt
+#
+# This file is part of PIConGPU.
+#
+# PIConGPU is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# PIConGPU is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with PIConGPU.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+
+##
+## This configuration file is used by PIConGPU's TBG tool to create a
+## batch script for PIConGPU runs. For a detailed description of PIConGPU
+## configuration files including all available variables, see
+##
+##                      docs/TBG_macros.cfg
+##
+
+
+#################################
+## Section: Required Variables ##
+#################################
+
+TBG_wallTime="1:00:00"
+
+TBG_gpu_x=1
+TBG_gpu_y=4
+TBG_gpu_z=1
+
+TBG_gridSize="128 512 128"
+TBG_steps="10000"
+TBG_movingWindow="-m"
+
+#################################
+## Section: Optional Variables ##
+#################################
+
+TBG_plugin_period="100"
+
+# png image output (electron density)
+TBG_pngYX="--e_png.period !TBG_plugin_period --e_png.axis yx --e_png.slicePoint 0.5 --e_png.folder pngElectronsYX"
+TBG_pngYZ="--e_png.period !TBG_plugin_period --e_png.axis yz --e_png.slicePoint 0.5 --e_png.folder pngElectronsYZ"
+
+# Create a particle-energy histogram [in keV] per species for every .period steps
+TBG_e_histogram="--e_energyHistogram.period !TBG_plugin_period --e_energyHistogram.binCount 1024 \
+                 --e_energyHistogram.minEnergy 0 --e_energyHistogram.maxEnergy 100000"
+
+# Calculate a 2D phase space
+# - requires parallel libSplash for HDF5 output
+# - momentum range in m_<species> c
+TBG_e_PSxpx="--e_phaseSpace.period !TBG_plugin_period --e_phaseSpace.space x --e_phaseSpace.momentum px --e_phaseSpace.min -8.0 --e_phaseSpace.max 8.0"
+TBG_e_PSypy="--e_phaseSpace.period !TBG_plugin_period --e_phaseSpace.space y --e_phaseSpace.momentum py --e_phaseSpace.min  0.0 --e_phaseSpace.max 1.0"
+TBG_e_PSzpz="--e_phaseSpace.period !TBG_plugin_period --e_phaseSpace.space z --e_phaseSpace.momentum pz --e_phaseSpace.min -2.0 --e_phaseSpace.max 2.0"
+
+TBG_plugins="!TBG_pngYX                    \
```

Comment:
QUESTION:
Why is this so much different to the other `*.cfg` files that it needs to be a separate file?
Couldn't we just use a combined `*.cfg`file to avoid duplicated code/pile up.

Reply:
We want to control steps directly from the outside, but our tools do not understand `-s number` but only tbg variables such as `!TBG_steps`, controlled with `tbg -o`.

We will also add probably more changes here, so it is best to make it a fully separate setup.

Reply:
Okay - that's a good idea :+1: